motion_class.py

```diff
@@ -0,0 +1,604 @@
+                for idx, foot in enumerate(self.motion_statics .foot_names)} for frame in range(self.n_frames)]
+                for motion in self.motion]
+
+    @property
+    def root_location(self) -> torch.tensor:
+        location = self.motion[..., :3, self.n_joints, :].detach().cpu()  # drop the 4th item in the position tensor
+        location = np.cumsum(location, axis=1) if self.use_velocity else location
+
+        return location  # K x T
+
+    def un_normalise(self, mean: torch.tensor, std: torch.tensor):
+        return self.sub_motion(self.motion * std + mean)
+
+    def sample_frames(self, frames_indexes: [int]):
+        return self.sub_motion(self.motion[..., frames_indexes])
+
+    def basic_anim(self):
+        offsets = self.motion_statics .offsets
+
+        positions = np.repeat(offsets[np.newaxis], self.n_frames, axis=0)
+        positions[:, 0] = self.root_location.transpose(0, 1)
+
+        orients = Quaternions.id(self.n_joints)
+        rotations = Quaternions(self.edge_rotations.permute(2, 1, 0).numpy())
+
+        if rotations.shape[-1] == 6:  # repr6d
+            from Motion.transforms import repr6d2quat
+            rotations = repr6d2quat(rotations)
+
+        anim_edges = Animation(rotations, positions, orients, offsets, self.motion_statics .parents)
+
+        return anim_edges
+
+    def move_rotation_values_to_parents(self, anim_exp):
+        children_all_joints = children_list(anim_exp.parents)
+        for idx, children_one_joint in enumerate(children_all_joints[1:]):
+            parent_idx = idx + 1
+            if len(children_one_joint) > 0:  # not leaf
+                assert len(children_one_joint) == 1 or (anim_exp.offsets[children_one_joint] == np.zeros(3)).all() and (
+                        anim_exp.rotations[:, children_one_joint] == Quaternions.id((self.n_frames, 1))).all()
+                anim_exp.rotations[:, parent_idx] = anim_exp.rotations[:, children_one_joint[0]]
+            else:
+                anim_exp.rotations[:, parent_idx] = Quaternions.id((self.n_frames))
+
+        return anim_exp
+
+    def anim_from_static(self):
+        anim_edges = self.basic_anim()
+
+        # TODO: Why do we need to sort names? isnt it already sorted correctly?
+        # sorted_order = get_sorted_order(anim_edges.parents)
+        # anim_edges_sorted = anim_edges[:, sorted_order]
+        # names_sorted = motion_statics .names['names_with_root'][sorted_order]
+
+        anim_exp, _, names_exp, _ = expand_topology_edges(anim_edges, names=self.motion_statics .names, nearest_joint_ratio=1)
+        anim_exp = self.move_rotation_values_to_parents(anim_exp)
+
+        return anim_exp, names_exp
+
+
+def plot(name):
+    def decorator(plot_method):
+        def wrapper(self, *args, **kwargs):
+            plt.figure()
+            plot_method(self, *args, **kwargs)
+            plt.title(name)
+            plt.savefig(f'fc/train_{name.replace(" ", "_")}')
+        return wrapper
+    return decorator
```

Comment:
when is plot() used? 
shouldn't it be in utils/visualization.py?

Reply:
Its a decorator for all the plot methods of the class DebugDynamic, So its used only for debug.
I added a doc line, but I am not sure this entire class of DebugDynamic is necessary at all.